wazimap/models/data.py

```diff
@@ -158,7 +158,11 @@ def get_db_table(self, release=None, year=None):
         if year is None and release is None:
             from wazimap.data.utils import current_context
             # use the current context
-            year = current_context().get('year')
+            try:
+                year = current_context().get('year')
```

Comment:
We haven't needed this up until now because all the views set the current_context() appropriately. Perhaps it would be better to ensure that the data API view sets the correct context and provides a default to 'latest' if the year isn't provided? This far down in the system, choosing 'latest' may not be the right thing to do.

Reply:
I wasn't too sure on this either. But the issue comes up when we call columns on a data table from a template.

https://github.com/OpenUpSA/wazimap/pull/128/files#diff-50ce8b3c3d1d7abb5f754060b4854f34L59

How do we ensure the context is correctly set there?

Setting it here doesn't seem to work:

https://github.com/OpenUpSA/wazimap/pull/128/files#diff-bb161a749d7997d3d3c7de1e8f3d11dcR378

Reply:
One solution is to set the columns in the view (enabling us to set the dataset_context) and pass it to the template:

https://github.com/OpenUpSA/wazimap/pull/128/commits/01458b3bc7d2f6ea5dfdf6126d089c297824b59b